aiida/tools/visualization/graph.py

```diff
@@ -0,0 +1,727 @@
+    if class_node_type == "data.int.Int.":
+        sublabel = "value: {}".format(node.get_attribute("value", ""))
+    elif class_node_type == "data.float.Float.":
+        sublabel = "value: {}".format(node.get_attribute("value", ""))
+    elif class_node_type == "data.str.Str.":
+        sublabel = "{}".format(node.get_attribute("value", ""))
+    elif class_node_type == "data.bool.Bool.":
+        sublabel = "{}".format(node.get_attribute("value", ""))
+    elif class_node_type == "data.code.Code.":
+        sublabel = "{}@{}".format(os.path.basename(node.get_execname()), node.get_computer_name())
+    elif class_node_type == "data.singlefile.SinglefileData.":
+        sublabel = node.filename
+    elif class_node_type == "data.remote.RemoteData.":
+        sublabel = "@{}".format(node.get_computer_name())
+    elif class_node_type == "data.structure.StructureData.":
+        sublabel = node.get_formula()
+    elif class_node_type == "data.cif.CifData.":
+        formulae = [str(f).replace(" ", "") for f in node.get_formulae() or []]
+        sg_numbers = [str(s) for s in node.get_spacegroup_numbers() or []]
+        sublabel_lines = []
+        if formulae:
+            sublabel_lines.append(", ".join(formulae))
+        if sg_numbers:
+            sublabel_lines.append(", ".join(sg_numbers))
+        sublabel = "; ".join(sublabel_lines)
+    elif class_node_type == "data.upf.UpfData.":
+        sublabel = "{}".format(node.get_attribute("element", ""))
+    else:
+        sublabel = node.get_description()
+
+    return sublabel
+
+
+def default_process_styles(node):
+    """map a process node to a graphviz node style
+
+    :param node: the node to map
+    :type node: aiida.orm.nodes.node.Node
+    :rtype: dict
+
+    """
+    class_node_type = node.class_node_type
+
+    if hasattr(node, "get_style_default"):
+        default = node.get_style_default()
+    else:
+        default = {"shape": "rectangle", "pencolor": "black"}
+
+    process_map = {
+        "process.calculation.calcjob.CalcJobNode.": {
+            "style": "filled",
+            "shape": "rectangle",
+            "pencolor": "black",
+            "fillcolor": "#87CEFA"  # light blue
+        },
+        "process.calculation.calcfunction.CalcFunctionNode.": {
+            "style": "filled",
+            "shape": "parallelogram",
+            "pencolor": "black",
+            "fillcolor": "#87CEFA"  # light blue
+        },
+        "process.workflow.workchain.WorkChainNode.": {
+            "style": "filled",
+            "shape": "octagon",
+            "pencolor": "black",
+            "fillcolor": "#FFA500"
+        },
+        "process.workflow.workfunction.WorkFunctionNode.": {
+            "style": "filled",
+            "shape": "hexagon",
+            "pencolor": "black",
+            "fillcolor": "#FFA500"  # light blue
+        }
+    }
+
+    node_style = process_map.get(class_node_type, default)
+
+    # style process node, based on success/failure of process
```

Comment:
> As per the discussion regarding default node styles, I think only shape is required to distinguish node types. 

It is certainly *possible* to use a different shape for each node but my personal impression (seems to be shared by sebastiaan and Giovanni) is that our need for very wide nodes makes the shapes other than ellipse and rectangle either unusable or very contorted.

> The default (white) is for processes that are not finised. Its also just more encouraging to see (hopefully) lots of green nodes :) 

While I share the sentiment, I think most of the time verdi graphs will be used to discuss finished calculations (e.g. when you want to show in a presentation how your workflow looks like). I.e. I think the "failed" state should be the one highlighted.

Reply:
> It is certainly possible to use a different shape for each node but my personal impression (seems to be shared by sebastiaan and Giovanni) is that our need for very wide nodes makes the shapes other than ellipse and rectangle either unusable or very contorted.

Indeed. Maybe the only difference, if you really want to have one, is to use rectangles with sharp or rounded edges between e.g. Calc and Work - but other shapes become less good for writing text in them.

One could think to an option to highlight the state of a process e.g. with the color of the edge (green for OK, yellow if being processed, red if failed), I don't know though if this should be on by default or this can just go in the textual description?

Reply:
> One could think to an option to highlight the state of a process e.g. with the color of the edge

Yes, that's what I suggested above as well.
I suggested to highlight only "failed" ones but since it's just the edge I would be fine with including it for any status as well.